cmd/clusterctl/client/rollout.go

```diff
@@ -20,53 +20,94 @@ import (
 	"fmt"
 	"strings"
 
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/util"
 )
 
-// RolloutRestartOptions carries the options supported by rollout restart.
-type RolloutRestartOptions struct {
+// RolloutOptions carries the base set of options supported by rollout command.
+type RolloutOptions struct {
 	// Kubeconfig defines the kubeconfig to use for accessing the management cluster. If empty,
 	// default rules for kubeconfig discovery will be used.
 	Kubeconfig Kubeconfig
 
-	// Resources to be rollout restarted.
+	// Resources for the rollout command
 	Resources []string
 
 	// Namespace where the resource(s) live. If unspecified, the namespace name will be inferred
 	// from the current configuration.
 	Namespace string
 }
 
-func (c *clusterctlClient) RolloutRestart(options RolloutRestartOptions) error {
+func (c *clusterctlClient) RolloutRestart(options RolloutOptions) error {
 	clusterClient, err := c.clusterClientFactory(ClusterClientFactoryInput{Kubeconfig: options.Kubeconfig})
 	if err != nil {
 		return err
 	}
+	tuples, err := getResourceTuples(clusterClient, options)
+	if err != nil {
+		return nil
```

Comment:
Umm...is this correct? Or should this be `return err`? It also looks like the tests all passed having this logic in so if this is indeed meant to be `return err` then we should have a unit test that asserts this. Thanks!

Reply:
Thanks for pointing that out! The fakeClient was not calling the individual funcs for the subcommands. 
I also had to separate these tests -- for example, the RolloutResume will throw an error if the md is not paused, while the RolloutRestart will do the opposite if it is paused, so the same test cases cannot be passed to all of them. 
There are additional tests in the lower level funcs that check for functional correctness. Is it okay if we keep these tests relatively simple?

Reply:
yeah...that's fine. I just wanted to ensure that if I changed this to `return nil` and run the tests in the `client` pkg, then I would see a failing test for the right reason. 🙂